dashboard/test/models/lesson_test.rb

```diff
@@ -935,4 +935,141 @@ def create_swapped_lockable_lesson
+      assert_equal 2, copied_lesson.resources.length
+      assert_equal @original_lesson.resources.map {|r| r.attributes.slice('name', 'url', 'properties').to_a}, copied_lesson.resources.map {|r| r.attributes.slice('name', 'url', 'properties').to_a}
+      assert_equal 2, copied_lesson.vocabularies.length
+      assert_equal @original_lesson.vocabularies.map(&:word), copied_lesson.vocabularies.map(&:word)
+      assert_equal 2, copied_lesson.objectives.length
+      assert_equal @original_lesson.objectives.map(&:description), copied_lesson.objectives.map(&:description)
+      assert_equal @original_lesson.standards, copied_lesson.standards
+      assert_equal @original_lesson.opportunity_standards, copied_lesson.opportunity_standards
+      assert_equal @original_lesson.programming_expressions, copied_lesson.programming_expressions
+    end
+
+    test "can clone lesson with duplicated resources and vocab into another script" do
+      create :resource, name: 'resource1', course_version: @original_course_version, lessons: [@original_lesson]
+      create :vocabulary, word: 'word one', course_version: @original_course_version, lessons: [@original_lesson]
+
+      @destination_script.expects(:write_script_json).once
+      destination_resource = create :resource, name: 'resource1', course_version: @destination_course_version
+      destination_vocab = create :vocabulary, word: 'word one', course_version: @destination_course_version
+      copied_lesson = Lesson.copy_to_script(@original_lesson, @destination_script)
+      assert_equal @destination_script, copied_lesson.script
+      assert_equal [destination_resource], copied_lesson.resources
+      assert_equal [destination_vocab], copied_lesson.vocabularies
+    end
+
+    test "can clone lesson into another script with lessons" do
+      lesson_activity = create :lesson_activity, lesson: @original_lesson
+      activity_section = create :activity_section, lesson_activity: lesson_activity
+      level1 = create :maze, name: 'level 1'
+      create :script_level, script: @original_script, lesson: @original_lesson, levels: [level1],
+        activity_section: activity_section, activity_section_position: 1
+
+      existing_lesson = create :lesson, script: @destination_script, lesson_group: @destination_lesson_group, has_lesson_plan: true
+      existing_lesson_activity = create :lesson_activity, lesson: existing_lesson
+      existing_activity_section = create :activity_section, lesson_activity: existing_lesson_activity
+      level2 = create :maze, name: 'level 2'
+      create :script_level, script: @destination_script, lesson: existing_lesson, levels: [level2],
+        activity_section: existing_activity_section, activity_section_position: 1
+
+      @destination_script.expects(:write_script_json).once
+      copied_lesson = Lesson.copy_to_script(@original_lesson, @destination_script)
+      @destination_script.reload
+      assert_equal @destination_script, copied_lesson.script
```

Comment:
this assertion verifies the lesson's script_id. it would be good to also verify the lesson's lesson_group_id.